archive_viewer/table_models/curve_model.py

```diff
@@ -48,7 +48,35 @@ def __contains__(self, key: str) -> bool:
         bool
             If the channel already exists in the model
         """
-        return key in [curve.address for curve in self._plot._curves]
+        for curve in self._plot._curves:
+            if hasattr(curve, "channel"):
+                if curve.address == key:
+                    return True
+            elif curve.formula == key:
+                return True
+        return False
+
+    def __contains__(self, key: str) -> bool:
```

Comment:
`__contains__` is duplicated for some reason, possibly as an artifact of a merge conflict

Reply:
They both seem to be the same so take your pick on which to delete

Reply:
Flipped a coin, it landed on its side. Decided to delete bottom half of the top one and top half of the bottom one. Hope this helps!